component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -2525,10 +2525,16 @@ private long getUnlockTimeInMilliSeconds(AuthenticatedUser authenticatedUser) th
     private void publishPostEmailOTPGeneratedEvent(HttpServletRequest request, AuthenticationContext context)
             throws AuthenticationFailedException {
 
-        Map<String, Object> eventProperties = new HashMap<>();
-        eventProperties.put(IdentityEventConstants.EventProperty.CORRELATION_ID, context.getCallerSessionKey());
         AuthenticatedUser authenticatedUser = (AuthenticatedUser) context.getProperty(EmailOTPAuthenticatorConstants
                 .AUTHENTICATED_USER);
+        Map<String, String> emailOTPParameters = getAuthenticatorConfig().getParameterMap();
+        if (isEmailOTPDisableForUser(authenticatedUser.getAuthenticatedSubjectIdentifier(),
```

Comment:
Shouldn't this check be done at the place where publishPostEmailOTPGeneratedEvent is called?

Reply:
Since publishPostEmailOTPGeneratedEvent() has been called twice in the process method, we have to repeat the logic in the both places. That's why this check is added inside this method. WDYT?